File: cpp/base/src/vsdk/toolkit/io/bin/reader/BinaryModelVertexRecordData.cpp
#include "vsdk/toolkit/io/bin/reader/BinaryModelVertexRecordData.h"

BinaryModelVertexRecordData::BinaryModelVertexRecordData():
    id(0),
    pointIndex(-1),
    normalIndex(-1),
    textureCoordinateIndex(-1),
    color(0.0, 0.0, 0.0),
    backIndex(-1),
    tmp(0),
    hasRadianceData(false),
    patchIndices()
{
}

File: java/base/src/main/java/vsdk/toolkit/io/bin/reader/BinaryModelVertexRecordData.java
package vsdk.toolkit.io.bin.reader;

import vsdk.toolkit.common.color.ColorRgbMutable;

public class BinaryModelVertexRecordData {
    public int id;
    public int pointIndex;
    public int normalIndex;
    public int textureCoordinateIndex;
    public ColorRgbMutable color;
    public int backIndex;
    public int tmp;
    public boolean hasRadianceData;
    public BinaryModelIndexListRef patchIndices;

    public BinaryModelVertexRecordData() {
        id = 0;
        pointIndex = -1;
        normalIndex = -1;
        textureCoordinateIndex = -1;
        color = new ColorRgbMutable();
        backIndex = -1;
        tmp = 0;
        hasRadianceData = false;
        patchIndices = new BinaryModelIndexListRef();
    }
}
